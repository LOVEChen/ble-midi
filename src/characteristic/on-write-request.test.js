/* eslint-env jest */
jest.mock('bleno/lib/characteristic');
jest.mock('../util/logger');
jest.mock('../packets/parse-packet');
jest.mock('../util/print-buffer');

let mockLog;
let mockParsePacket;
let mockOnIncomingPacket;
let onWriteRequest;

beforeEach(() => {
  jest.resetAllMocks();
  mockLog = require('../util/logger').default;
  mockParsePacket = require('../packets/parse-packet').default;
  mockOnIncomingPacket = jest.fn();
  onWriteRequest = require('./on-write-request').default;
});

test('should onWriteRequest(validData, 0, true, callback)', () => {
  const data = new Buffer([1, 2, 3, 4, 5]);
  const offset = 0;
  const withoutResponse = true;
  const callback = jest.fn();
  onWriteRequest(mockOnIncomingPacket, data, offset, withoutResponse, callback);
  expect(mockLog.mock.calls).toMatchSnapshot();
  expect(mockLog.mock.calls.length).toEqual(2);
  expect(mockLog.mock.calls[0].length).toEqual(1);
  expect(callback.mock).toMatchSnapshot();
  expect(callback.mock.calls.length).toEqual(0);
  expect(mockParsePacket.mock.calls).toMatchSnapshot();
  expect(mockOnIncomingPacket.mock).toMatchSnapshot();
});

test('should RESULT_UNLIKELY_ERROR onWriteRequest withoutResponse', () => {
  const data = new Buffer([1, 2, 3, 4, 5]);
  const callback = jest.fn();
  onWriteRequest(mockOnIncomingPacket, data, 0, false, callback);
  expect(mockLog.mock.calls).toMatchSnapshot();
  expect(callback.mock).toMatchSnapshot();
  expect(callback.mock.calls.length).toEqual(1);
  expect(mockOnIncomingPacket.mock).toMatchSnapshot();
});

test('should RESULT_UNLIKELY_ERROR onWriteRequest with invalid offset', () => {
  const data = new Buffer([1, 2, 3, 4, 5]);
  const callback = jest.fn();
  onWriteRequest(mockOnIncomingPacket, data, 5, true, callback);
  expect(mockLog.mock.calls).toMatchSnapshot();
  expect(callback.mock).toMatchSnapshot();
  expect(callback.mock.calls.length).toEqual(1);
  expect(mockOnIncomingPacket.mock).toMatchSnapshot();
});

test('should RESULT_UNLIKELY_ERROR onWriteRequest with invalid data', () => {
  const invalidData = new Buffer(0);
  const callback = jest.fn();
  onWriteRequest(mockOnIncomingPacket, invalidData, 0, true, callback);
  expect(mockLog.mock.calls).toMatchSnapshot();
  expect(callback.mock).toMatchSnapshot();
  expect(callback.mock.calls.length).toEqual(1);
});
