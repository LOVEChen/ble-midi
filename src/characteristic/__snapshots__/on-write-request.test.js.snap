// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should RESULT_UNLIKELY_ERROR onWriteRequest with invalid data 1`] = `
Array [
  Array [
    "onWriteRequest:
             data: 

                   undefined
      data.length: 0
           offset: 0
  withoutResponse: true
  ",
  ],
]
`;

exports[`should RESULT_UNLIKELY_ERROR onWriteRequest with invalid data 2`] = `
Object {
  "calls": Array [
    Array [
      14,
    ],
  ],
  "instances": Array [
    undefined,
  ],
}
`;

exports[`should RESULT_UNLIKELY_ERROR onWriteRequest with invalid offset 1`] = `
Array [
  Array [
    "onWriteRequest:
             data: 0102030405

                   undefined
      data.length: 5
           offset: 5
  withoutResponse: true
  ",
  ],
]
`;

exports[`should RESULT_UNLIKELY_ERROR onWriteRequest with invalid offset 2`] = `
Object {
  "calls": Array [
    Array [
      14,
    ],
  ],
  "instances": Array [
    undefined,
  ],
}
`;

exports[`should RESULT_UNLIKELY_ERROR onWriteRequest with invalid offset 3`] = `
Object {
  "calls": Array [],
  "instances": Array [],
}
`;

exports[`should RESULT_UNLIKELY_ERROR onWriteRequest withoutResponse 1`] = `
Array [
  Array [
    "onWriteRequest:
             data: 0102030405

                   undefined
      data.length: 5
           offset: 0
  withoutResponse: false
  ",
  ],
  Array [
    "parsedPacket",
    undefined,
  ],
]
`;

exports[`should RESULT_UNLIKELY_ERROR onWriteRequest withoutResponse 2`] = `
Object {
  "calls": Array [
    Array [
      14,
    ],
  ],
  "instances": Array [
    undefined,
  ],
}
`;

exports[`should RESULT_UNLIKELY_ERROR onWriteRequest withoutResponse 3`] = `
Object {
  "calls": Array [
    Array [
      undefined,
    ],
  ],
  "instances": Array [
    undefined,
  ],
}
`;

exports[`should onWriteRequest(validData, 0, true, callback) 1`] = `
Array [
  Array [
    "onWriteRequest:
             data: 0102030405

                   undefined
      data.length: 5
           offset: 0
  withoutResponse: true
  ",
  ],
  Array [
    "parsedPacket",
    undefined,
  ],
]
`;

exports[`should onWriteRequest(validData, 0, true, callback) 2`] = `
Object {
  "calls": Array [],
  "instances": Array [],
}
`;

exports[`should onWriteRequest(validData, 0, true, callback) 3`] = `
Array [
  Array [
    Object {
      "data": Array [
        1,
        2,
        3,
        4,
        5,
      ],
      "type": "Buffer",
    },
  ],
]
`;

exports[`should onWriteRequest(validData, 0, true, callback) 4`] = `
Object {
  "calls": Array [
    Array [
      undefined,
    ],
  ],
  "instances": Array [
    undefined,
  ],
}
`;
